src/main/services/opds.ts

```diff
@@ -151,6 +151,7 @@ export class OpdsService {
                 if (url.search.includes(SEARCH_TERM) ||
                     tryDecodeURIComponent(url.pathname).includes(SEARCH_TERM)) {
 
+                    atomLink.url = atomLink.url.replace("%7B", "{").replace("%7D", "}")
```

Comment:
The `String.replace()` function with a string function argument replaces only the first occurrence. I think you need to use a regular expression to "replace all":

`atomLink.url = atomLink.url.replace(/%7B/g, "{").replace(/%7D/g, "}")`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace

Also, there is a missing trailing semicolon (lint)

Reply:
I wonder why such an ad-hoc string replacement is needed. As we haven't come across this problem in our tests, I am concerned about introducing a regression bug. Normally this would be handled by `decodeURIComponent()`. Do you have an example of Atom URL that exemplifies the problematic syntax?